### ISSUE TITLE
Please reactivate removed repo (orocrm/composer)

### ISSUE THREAD

**Issue by sbani** [oroinc/platform#779]
Hi,

we are still using `oro/platform@v1.9.3` which has a dependency on `composer/composer@v1.0.0-p1` ([here](https://github.com/oroinc/platform/blob/1.9.3/composer.json#L61)).

Unfortunately, you removed the repository.
**Could you please reactivate this repo? Is it possible to use one of the official newer composer versions?**

We would host it on our github. Perhaps you can send us the code?

Best regards

**Comment by aivus** [comment]
Hi @sbani 

Do you mean this repo https://github.com/oroinc/composer ?

**Comment by DimaSoroka** [comment]
Hi @sbani 

Repository was restored, 1.9.3 is very old version of the platform, please consider upgrade to a newer one.

**Comment by sbani** [comment]
Thank you @DimaSoroka. I know, but we are not able to upgrade for now.

**Comment by danbentley** [comment]
I think this still might be an issue. Our composer still fails on install:

 ``` 
- Installing composer/composer (1.0.0-p1): Downloading (failed)    Failed to download composer/composer from dist: The "https://api.github.com/repos/orocrm/composer/zipball/833ce984264204e7d6576ab082660105c7d8f04c" file could not be downloaded (HTTP/1.1 404 Not Found)
    Now trying to download from source
  - Installing composer/composer (1.0.0-p1): Cloning 833ce98426

  [RuntimeException]
  Failed to execute git clone --no-checkout 'https://***:***@github.com/orocrm/composer.git' '/var/www/sites/ca1/www/vendor/composer/composer' && cd '/var/
  www/sites/ca1/www/vendor/composer/composer' && git remote add composer 'https://***:***@github.com/orocrm/composer.git' && git fetch composer
  Cloning into '/var/www/sites/ca1/www/vendor/composer/composer'...
  remote: Repository not found.
  fatal: repository 'https://***:***@github.com/orocrm/composer.git/' not found
```

It looks like https://packagist.orocrm.com/packages/composer/composer is still referencing the old repo: `github.com/orocrm/composer.git`

**Comment by sbani** [comment]
@danbentley, add this to you `composer.json`:

```
    "repositories": [
...
        {
            "type": "vcs",
            "url": "https://github.com/oroinc/composer.git"
        }
...
```

**Comment by danbentley** [comment]
Thanks @sbani.

For anyone else with the same issue, it appears the order of entries in the `"repositories"` array is significant. This fixed it for us:

```
…
"repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/oroinc/composer"
        },
        {
            "type": "composer",
            "url": "https://packagist.orocrm.com"
        }
    ],
…
```

**Comment by dave08** [comment]
I have the same problem and tried all of the above to no avail... is there anything else to do about this?

**Comment by mbessolov** [comment]
@dave08  what version of OroPlatform do you have?

**Comment by dave08** [comment]
@mbessolov I have 1.9.3, but don't have time to update to 1.10... I also just tried running composer update and it gives me:
```
oro/platform 1.9.3 requires sensio/distribution-bundle 4.0.3 -> satisfiable by sensio/distribution-bundle[v4.0.3] but these conflict with your requirements or minimum-stability.
```

My Jenkins pipeline (while generating docker image) gives me (I'm using a github token...):
```
  - Installing composer/composer (1.0.0-p1)

    Downloading: Connecting...    Downloading: Failed       

    Failed to download composer/composer from dist: The "https://api.github.com/repos/orocrm/composer/zipball/833ce984264204e7d6576ab082660105c7d8f04c" file could not be downloaded (HTTP/1.1 404 Not Found)

    Now trying to download from source

  - Installing composer/composer (1.0.0-p1)

    Cloning 833ce984264204e7d6576ab082660105c7d8f04c


                                                                                              

  [RuntimeException]                                                                          

  Failed to clone https://github.com/orocrm/composer.git via https, ssh protocols, aborting.  

  - https://github.com/orocrm/composer.git                                                    

    Cloning into '/app/vendor/composer/composer'...                                           

    remote: Invalid username or password.                                                     

    fatal: Authentication failed for 'https://github.com/orocrm/composer.git/'                

  - git@github.com:orocrm/composer.git                                                        

    Cloning into '/app/vendor/composer/composer'...                                           

    Host key verification failed.                                                             

    fatal: Could not read from remote repository.                                             

                                                                                              

    Please make sure you have the correct access rights                                       

    and the repository exists.       
```

**Comment by mbessolov** [comment]
@dave08 try to change the URL to https://github.com/oroinc/composer
orocrm org has been renamed to oroinc quite a while ago and apparently GitHub does not maintain redirects that long.

**Comment by dave08** [comment]
@mbessolov This is my repositories:
```
    "repositories": [
        {
            "type": "vcs",
            "no-api": true,
            "url": "https://github.com/oroinc/composer.git"
        },
        {
            "type": "vcs",
            "no-api": true,
            "url": "https://github.com/oroinc/composer"
        },
        {
            "type": "composer",
            "url": "https://packagist.orocrm.com"
        },
        {
            "type": "composer",
            "url": "https://packagist.orocrm.com"
        }
    ],
```

**Comment by aivus** [comment]
@mbessolov @DimaSoroka could we transfer this repo to `orocrm` organization and then return to `oroinc`? It creates all required redirects.

**Comment by mbessolov** [comment]
@aivus It will wipe the repository permissions and somebody will have to spend a few hours to return them back to the original state, which is too much for an unsupported version. And it is not necessary, as the application installs with a simple modification to composer.lock.

@dave08 what you posted looks like a fragment of composer.json, and this not the same composer.json that is included in oroinc/platform-application 1.9.3 ([original composer.json](https://github.com/oroinc/platform-application/blob/1.9.3/composer.json))
You may not need any of those new repositories to you **composer.json**

You need to modify [2 lines (537, 542)](https://github.com/oroinc/platform-application/blob/1.9.3/composer.lock#L537-L542) in **composer.lock**. Replace **/orocrm/composer/** with **/oroinc/composer/** and run composer install again. Please note that the line numbers may be different, so just look for all references to **orocrm/composer** in your composer.lock
